apf/apf.h

```diff
@@ -713,12 +718,15 @@ void unfreeze(Field* f);
 /** \brief Returns true iff the Field uses array storage. */
 bool isFrozen(Field* f);
 
+template <typename T>
+T* getArrayDataT(Field * f);
```

Comment:
Should this go into the cc file and use template specializations, as to not confuse the user space interface? If not possibly add a comment mentioning that this shouldn't be directly called by user?

Reply:
Yeah that's probably the way to go. Don't necessarily want users exposed to this template function. Wrapping up use of the template function in a couple of interface functions similar to how `getArrayData` is now implemented is probably the best short-term solution. 

Unfortunately there is a potential for misuse if a user supplies a 'Field' to one of these functions with the wrong underlying type, since we basically have to rely on a `reinterpret_cast` to get the underlying data of the different type. I'll see if I can add an assert or something to try to prevent this issue.